src/installation/guides/zfs.md

```diff
@@ -0,0 +1,330 @@
+
+`(chroot)# zpool set bootfs=zroot/ROOT/void zroot`
+
+Also ensure GRUB recognizes the ZFS module.
+
+> **Note:**
+> 
+> When using id's set the ZPOOL_VDEV_NAME_PATH variable to 1.
+
+```
+(chroot)# ZPOOL_VDEV_NAME_PATH=1 grub-probe /
+zfs
+```
+
+> **Optional:**
+> 
+
+Install GRUB onto the drive specifying the path to the drive:
+
+```
+(chroot)# ZPOOL_VDEV_NAME_PATH=1 grub-install /dev/sda
+Installing for i386-pc platform.
+Installation finished. No error reported.
+```
+
+Create and modify `/etc/dracut.conf.d/zol.conf` to configure Dracut to include
+at least the following values:
+
+```
+hostonly="yes"
+nofsck="yes"
+add_dracutmodules+=" zfs "
+```
+
+Reconfigure Linux to update GRUB's and Dracut's configurations:
+
+> **Note:**
+> 
+> If the Linux version on the system is unknown use a query to find the
+> installed version on the new system:
+> 
+> `xbps-query -s linux | grep modules`
+
+```
+(chroot)# xbps-reconfigure -f linux5.4
+linux5.4: configuring ...
+Executing post-install kernel hook: 10-dkms ...
+Available DKMS module: zfs-0.8.3.
+Building DKMS module: zfs-0.8.3... done.
+Executing post-install kernel hook: 20-dracut ...
+Executing post-install kernel hook: 50-grub ...
+Generating grub configuration file ...
+Found linux image: /boot/vmlinuz-5.4.32_1
+Found initrd image: /boot/initramfs-5.4.32_1.img
+Found void on /dev/sda2
+done
+linux5.4: configured successfully.
+```
+
+## Sanity Checks
+
+Ensure that the Dracut configuration took with
+[lsinitrd(1)](https://man.voidlinux.org/lsinitrd):
+
+> **Note:**
+> 
+> It may be necessary to specify the path to the img if the kernel version has
+> changed, see [lsinitrd(1)](https://man.voidlinux.org/lsinitrd).
+
+```
+(chroot)# lsinitrd -m
+initrd in UEFI: : 13M
+========================================================================
+Version: 
+
+dracut modules:
+bash
+dash
+i18n
+drm
+kernel-modules
+kernel-modules-extra
+zfs
+resume
+rootfs-block
+terminfo
+udev-rules
+usrmount
+base
+fs-lib
+shutdown
+========================================================================
+```
+
+As well check that the ZFS cache is recognized:
+
+```
+(chroot)# lsinitrd | grep zpool.cache
+-rw-r--r--   1 root     root         1376 Apr 17 01:30 etc/zfs/zpool.cache
+```
+
+## Cleanup
+
+Exit chroot:
+
+`(chroot)$ exit`
+
+And un-mount the chroot environment:
+
+```
+# umount /mnt/void/dev/pts
+# umount /mnt/void/{dev,proc,sys}
+# zpool export zroot
+```
+
+Restart the system using [reboot(8)](https://man.voidlinux.org/reboot) and log
+in:
+
+`# reboot`
+
+## Configuration
+
+Configure the defaults in the new system's `/etc/rc.conf`. Below are some values
+that most system's will want to set.
+
+````````
+# see loadkeys(8)
+KEYMAP="us"
+# available timezones at /usr/share/zoneinfo
+TIMEZONE="America/Chicago"
+# set to UTC or localtime
+HARDWARECLOCK="UTC"
+````````
+
+Finally configure the system's locales consulting the
+[Locales](./../../config/locales.md) page of the handbook.
```

Comment:
Configuration should probably happen before cleanup, and should include setting the timezone with an `ln` instead of the `TIMEZONE` variable. I'd rather not duplicate rc.conf's comments here at all, there is a separate doc page for that.